src/figuresClasses.ts

```diff
@@ -1,11 +1,78 @@
-export interface Figure {}
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (
+      Math.min(a, b, c) <= 0 ||
+      a + b + c - Math.max(a, b, c) <= Math.max(a, b, c)
+    ) {
+      throw new Error('your error message');
+    }
+  }
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2; // semiperimetr
```

Comment:
variable name is not ok. If you fix it you won't ned the comment
```suggestion
    const semiperimetr = (this.a + this.b + this.c) / 2;
```